### PR TITLE
Make completeWithSuccess an async function

### PR DESCRIPTION
**Pull request by NotWoods**
## Description

Updates functions that use `Network.completeWithSuccess` to either work with async RxJava objects or be marked with `@WorkerThread`. 

## Additional context

This is part of work to move logic off the main thread.

### CODE REVIEW

**Comment by bekuno** [comment]
retest this please

**Comment by bekuno** [comment]
retest this please

**Comment by bekuno** [comment]
Single test error:
~~~
20:40:00  cgeo.geocaching.connector.gc.GCParserTest > testEditModifiedCoordinates[android-30(AVD) - 11] [31mFAILED [0m
20:40:00  	org.junit.ComparisonFailure: expected:<[tru]e> but was:<[fals]e>
20:40:00  	at java.lang.reflect.Constructor.newInstance0(Native Method)
~~~

File: main/src/main/java/cgeo/geocaching/connector/gc/GCParser.java [hunk 1]
```diff
@@ -1932,20 +1950,22 @@ static boolean editModifiedCoordinates(@NonNull final Geocache cache, final Geop
...
-            return false;
-        }
+        return Network.completeWithSuccess(Network.postJsonRequest(uriPrefix + uriSuffix, jo))
+            .toSingle(() -> {
+                Log.i("GCParser.editModifiedCoordinates - edited on GC.com");
+                return true;
+            })
+            .onErrorReturn((throwable) -> {
+                Log.e("GCParser.deleteModifiedCoordinates - cannot delete modified coords", throwable);
+                return false;
```
Comment (bekuno) [inline main/src/main/java/cgeo/geocaching/connector/gc/GCParser.java:1960]: On other places you changed to `return Single.just(false);` (or `true`).
Should here also, isn´t it?

File: main/src/main/java/cgeo/geocaching/connector/gc/GCParser.java [hunk 2]
```diff
@@ -1955,14 +1975,15 @@ static boolean uploadPersonalNote(@NonNull final Geocache cache) {
...
-            return false;
-        }
+        return Network.completeWithSuccess(Network.postJsonRequest(uriPrefix + uriSuffix, jo))
+            .toSingle(() -> {
+                Log.i("GCParser.uploadPersonalNote - uploaded to GC.com");
+                return true;
+            })
+            .onErrorReturn((throwable) -> {
+                Log.e("GCParser.uploadPersonalNote - cannot upload personal note", throwable);
+                return false;
```
Comment (bekuno) [inline main/src/main/java/cgeo/geocaching/connector/gc/GCParser.java:1985]: Same as before.

**Review by bekuno — CHANGES_REQUESTED** [review]

File: main/src/main/java/cgeo/geocaching/connector/gc/GCParser.java (hunk 1 above)
Comment (NotWoods) [reply main/src/main/java/cgeo/geocaching/connector/gc/GCParser.java:1960]: on Error Return expects a value rather than Single of value

**Review by bekuno — COMMENTED** [review]
Someone should check this PR.
I'm not able to do this.

**Comment by moving-bits** [comment]
> Someone should check this PR. I'm not able to do this.

Me neither, not my area of expertise.

**Comment by eddiemuc** [comment]
I believe changes to be uncritical. However PR has merge conflicts with master now

**Comment by bekuno** [comment]
@NotWoods Please resolve the conflict. Sorry that the merge was not done in the meantime.
@eddiemuc Please check the PR after the conflict is resolved.

**Pull request merged**